core/src/main/java/me/lokka30/treasury/plugin/core/command/subcommand/economy/migrate/MigrationEconomy.java

```diff
@@ -32,7 +33,82 @@
     private final @NotNull EconomyException migrationException;
 
     MigrationEconomy() {
-        this.currency = Currency.of(null, 0, 1, (amt, $) -> String.valueOf(amt), "MigrationMoney");
+        this.currency = new Currency() {
+            @Override
+            public String identifier() {
+                return "MigrationMoney";
+            }
+
+            @Override
+            public String symbol() {
+                return "$";
+            }
+
+            @Override
+            public char decimal() {
+                return 0;
+            }
+
+            @Override
+            public String display() {
+                return "MigrationMoney";
+            }
+
+            @Override
+            public String displayPlural() {
+                return "MigrationMonies";
+            }
+
+            @Override
+            public int precision() {
+                return 0;
+            }
+
+            @Override
+            public boolean isDefault() {
+                return false;
+            }
+
+            @Override
+            public boolean isDefault(@NotNull final String world) {
+                return false;
+            }
+
+            @Override
+            public void to(
+                    @NotNull final Currency currency,
+                    @NotNull final Double amount,
+                    @NotNull final EconomySubscriber<Double> subscription
+            ) {
+                subscription.fail(new EconomyException(FailureReason.MIGRATION, "Migration currency not convertable."));
+            }
+
+            @Override
+            public void deformat(@NotNull final String formatted, @NotNull final EconomySubscriber<Double> subscription) {
+                try {
+                    Double value = Double.parseDouble(formatted);
+                    subscription.succeed(value);
+                } catch(Exception ignore) {
+                    subscription.fail(new EconomyException(FailureReason.OTHER_FAILURE, "Formatted is not a valid Double value" +
+                            "."));
+                }
+            }
```

Comment:
This should always fail

Reply:
> I don't like the idea of removing the CurrencyManager and moving the currency methods to the EconomyProvider, we also lose the hability to parse any string into a curremcy and an amount

Could you elaborate on this? Also forcing that currency manager isn't a good practice as it has no business being in an API. An API should simply provide the connection for the actions to happen, not do the action for them.

Reply:
> This should always fail

Will fix this.

Reply:
> Could you elaborate on this

Ideally a class should only be in charge of 1 things. EconomyProvider is already in charge of Accounts and the CurrencyManager handled the currencies, merging them gives the EconomyProvider all the responsability.

We discussed this here anyways
https://github.com/lokka30/Treasury/issues/44
i suggested this but it seems like it went unnoticed
> About splitting the provider into several providers, could we have all providers splited as described by Jikoo or something like that and then have the economy provider extend all those interfaces?
This way if you want to use the separated parts them you just cast, if you want all use the EconomyProvider. This would allow also for the implementation to have its logic separated _by default_

However a EconomyProvider implementation could just delegate the tasks to other objects, kinda like the Bukkit class

So it comes down to my preferences

Reply:
> > Could you elaborate on this
> 
> Ideally a class should only be in charge of 1 things. EconomyProvider is already in charge of Accounts and the CurrencyManager handled the currencies, merging them gives the EconomyProvider all the responsability.
> 
> We discussed this here anyways #44 i suggested this but it seems like it went unnoticed
> 

Except for the fact that EconomyProvider still provided methods for the primary currency. This goes against the very essence of the class's documented description which is for "managing economy data" as outlined in the javadocs of the class. In essence if it should simply be for account management it shouldn't be called an EconomyProvider and shouldn't have the methods for supportedAPIVersion or supportedFeatures. Hence why CurrencyManager makes no sense to have when currencies are data for economies.

Reply:
You're right there

Reply:
> You're right there

You're not wrong though it really comes down to preference, which isn't really where an API should come into place.